src/form/field/form-field.test.tsx

```diff
@@ -0,0 +1,119 @@
+import React from "react";
+import {render, cleanup} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import {create} from "react-test-renderer";
+
+import {testA11y} from "../../core/utils/test/testUtils";
+import FormField, {FormFieldProps} from "./FormField";
+import Input from "../input/Input";
+
+describe("<FormField />", () => {
+  afterEach(cleanup);
+
+  const defaultFormFieldProps: FormFieldProps = {
+    testid: "form-field",
+    children: <Input onChange={jest.fn} name={"test"} testid={"form-field.input"} />
+  };
+
+  it("should render correctly", () => {
+    render(<FormField {...defaultFormFieldProps} />);
+  });
+
+  it("should matches snapshot", () => {
+    const tree = create(<FormField {...defaultFormFieldProps} />).toJSON();
+
+    expect(tree).toMatchSnapshot();
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<FormField {...defaultFormFieldProps} />);
+
+    await testA11y(container, {rules: {label: {enabled: false}}});
+  });
+
+  it("should render children correctly", () => {
+    const {getByTestId} = render(<FormField {...defaultFormFieldProps} />);
+
+    const childrenContent = getByTestId("form-field.input");
+
+    expect(getByTestId("form-field")).toContainElement(childrenContent);

```

Comment:
Should we get the testid from `defaultFormFieldProps` ? If so, let's do it for other uses below as well. Also, should we add a test to see if the data-testid is added as an attribute when testid prop is provided or is that handled with the snapshot ?
```suggestion
    expect(getByTestId(defaultFormFieldProps.testid)).toContainElement(childrenContent);
```

Reply:
- Yes, we can get the testid from default props. 🙏 
-  I think we don't need to test `testid` because snapshot test already handle it.